### PR TITLE
expression: simplify get priority by collation

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

simplify code

### What is changed and how it works?

What's Changed:

- make collation priority start from 1. use 0 indicate collation not exist in map
- rename varible `collationPriority` to `priority`, make it not confuse to map `collationPriority`

How it works:

- `collationPriority[collation]` will return 0 if collation is not in map. 
see https://golang.org/ref/spec#Index_expressions
see https://golang.org/ref/spec#The_zero_value

### Test
- Unit test
- Integration test

### Release note 

- No release note

### CODE REVIEW

File: expression/collation.go [hunk 1]
```diff
@@ -114,13 +114,13 @@ var (
 
 	// collationPriority is the priority when infer the result collation, the priority of collation a > b iff collationPriority[a] > collationPriority[b]
 	collationPriority = map[string]int{
-		charset.CollationASCII:   0,
```
Comment (wjhuang2016) [inline expression/collation.go:117]: I think we need to rewrite the whole function. Before unicode_ci, the assumption in the comment holds. After introducing unicode_ci, it doesn't hold anymore.

Comment (xiongjiwei) [reply expression/collation.go:117]: yes, `unicode_ci` and `general_ci`  is incompatible, my solution is make `unicode_ci` and `general_ci` has same priority. Same priority in `collationPriority ` means they are incompatible, and we can use [rule No.2](https://pingcap.com/docs-cn/stable/character-set-and-collation/#%E8%A1%A8%E8%BE%BE%E5%BC%8F%E4%B8%AD%E6%8E%92%E5%BA%8F%E8%A7%84%E5%88%99%E7%9A%84-coercibility-%E5%80%BC). I will rewrite `checkIllegalMixCollation` `deriveCoercibilityForScarlarFunc` and `DeriveCollationFromExprs`. This is first step to simplify code

**Pull request merged**